### PR TITLE
Update documentation — UG Autotags and DG design of Model

### PR DESCRIPTION
**Pull request by dorcastan**
Changes:
* User Guide
    * Updated *Autotags* section
    * Wrote overview of *Features* section
    * Moved command parameters table to *Command Summary*
* Developer Guide
    * Updated class diagrams for *Model* section (Design)
* Miscellaneous
    * Added JSoup and Readability4J to the libraries used in `README.md`
    * Deleted `johndoe.adoc`

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -522,78 +524,135 @@ Examples:
...
 
 Format: `*autotag-delete* TAG_NAME`
 
+For example:
+
+* If you want to delete the tag `NUS`, input `*autotag-delete* NUS` into the command box.
++
+image::ui-screenshots/AutotagDeleteCommandUi1.png[]
+
+* Observe that the autotag named `NUS` has been deleted from the autotag panel of the Dashboard.
```
Comment (TSAI-HSIAO-HAN) [inline docs/UserGuide.adoc]: Prevent to use command sentence. (from 2101)

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -1035,33 +1099,67 @@ General notes are found at the bottom of the rightmost column on the offline cop
...
+| Delete cache: `*cache-delete* INDEX`
+| Offline view: `*offline* INDEX`
+| Add annotation: `*annotate* INDEX p/P_NUM [n/NOTES] [h/HIGHLIGHT_COLOUR=yellow]`
+| Edit annotation: `*annotate-edit* INDEX p/P_NUM [to/NEW_P_NUM] [n/NOTES] [h/HIGHLIGHT_COLOUR]`
+| Delete annotation: `*annotate-delete* INDEX [p/P_NUM] [n/KEEP_NOTES=false] [h/KEEP_HIGHLIGHT=false]`
+
+.3+|Reminders
+| Add reminder: `*reminder* INDEX t/TIME_DATE [n/NOTE=Open]`
+| Edit reminder: `*reminder-edit* INDEX [t/TIME_DATE] [n/NOTE]`
+| Delete reminder: `*reminder-delete* INDEX`
```
Comment (TSAI-HSIAO-HAN) [inline docs/UserGuide.adoc]:
```suggestion
| Delete reminder: `*reminder-delete* INDEX`
| Goto reminder: `*reminder-goto* INDEX`
```

File: docs/diagrams/ModelClassDiagram.puml [hunk 3]
```diff
@@ -54,42 +47,39 @@ Class HiddenOutside #FFFFFF
...
+
+FolderStructure -[hidden]left-> Autotag
+Autotag -[hidden]left-> Reminder
+
+'UniqueBookmarkList o--> Bookmark
+Bookmark *-left-> Tag
+Bookmark --> Annotation
+
+HiddenB1 -[hidden]right-> HiddenB2
+
```
Comment (TSAI-HSIAO-HAN) [inline docs/diagrams/ModelClassDiagram.puml]: Should be okay

File: docs/UserGuide.adoc (hunk 2 above)
Comment (dorcastan) [reply docs/UserGuide.adoc]: I added it as `*goto-reminder*` based on the current implementation!

File: docs/UserGuide.adoc (hunk 1 above)
Comment (dorcastan) [reply docs/UserGuide.adoc]: @kxyoke I changed the format to "You can then check that [result of command, etc.]" but didn't touch the corresponding parts in the annotation sections! Do you want me to edit it, or you'll do it?

File: docs/UserGuide.adoc [hunk 4]
```diff
@@ -522,78 +524,135 @@ Examples:
...
-This command deletes a previously created <<autotag, autotag>> from Mark. Tags that are already present will
-not be removed, but no bookmarks will be tagged by this autotag after it is deleted.
+If you no longer need an autotag, you can delete it from Mark using the `autotag-delete`
+command. None of your existing tags are affected when an autotag is deleted.
 
 Format: `*autotag-delete* TAG_NAME`
 
+For example:
+
+* If you want to delete the tag `NUS`, input `*autotag-delete* NUS` into the command box.
```
Comment (openorclose) [inline docs/UserGuide.adoc]:
```suggestion
* If you want to delete the autotag `NUS`, input `*autotag-delete* NUS` into the command box.
```

should this be autotag?

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -165,15 +166,41 @@ when the list data are changed.
 Mark. Each `Mark`, in turn, comprises a `UniqueBookmarkList`, a `ReminderAssociation` object, an
 `AutotagController`, and a `FolderStructure`.
 
+.Structure of the Bookmark package
```
Comment (openorclose) [inline docs/DeveloperGuide.adoc]:
```suggestion
.Structure of the Bookmark Package
```

capitalize like all the other captions?

Comment (openorclose) [reply docs/DeveloperGuide.adoc]: actually nvm i looked at all the captions and theyre not consistent anyway, we can just fix it all later

Comment (openorclose) [reply docs/DeveloperGuide.adoc]: i think we should all switch to sentence case, since some captions quite long

File: docs/UserGuide.adoc (hunk 2 above)
Comment (TSAI-HSIAO-HAN) [reply docs/UserGuide.adoc]: I'll do it.

File: docs/DeveloperGuide.adoc (hunk 5 above)
Comment (dorcastan) [reply docs/DeveloperGuide.adoc]: Mm okay. Will do that later.

File: docs/UserGuide.adoc [hunk 6]
```diff
@@ -379,17 +380,17 @@ Other examples:
 * `*add* u/https://nus-cs2103-ay1920s1.github.io/website/ n/Module Website
 f/CS2103T r/Contains textbook & important deadlines`
 +
-Bookmarks the given website, names it `Module Website`, and adds it to the folder `CS2103T`.
-The new bookmark's remark is `Contains textbook & important deadlines`.
+image::ui-screenshots/AddCommandExample1.png[]
 +
-image::AddCommandExample1.png[]
+Bookmarks the given website, names it `Module Website`, and adds it to the folder `CS2103T`.
```
Comment (TSAI-HSIAO-HAN) [inline docs/UserGuide.adoc]: Prevent using command sentence.

**Pull request merged**